docs/designs/flatten_odps_key_value_table.md

```diff
@@ -0,0 +1,107 @@
+# Transform the Column with Key-value Pairs String to Wide Table
+This document describes the design to transform the column with key-value pairs string to a wide ODPS table where
+the values of each key is in a each column.
+
+## Motivation
+Generally, the data developers parse logs to features and the number of features may vary. It is convenient to save features using
+key-value pair string as a column in a table. However, it is difficult to analyze features in the table using SQL syntax. So we need to transform
+the column to a wide table. Then we can analyze each features like calculate the mean and variance for the numeric feature. For example, the key-value column is
+
+| features| label|
+|---------|------|
+| age:19,income:1200,education:Master | 1 |
+| age:23,income:4500,education:Doctor | 0 |
+
+And we need to transform it to 
+
+| age | income | education | label |
+|-----|--------|-----------|-------|
+| 19  | 1200   | Master    | 1 |
+| 23  | 4500   | Doctor    | 0 |
+
+
+## Design Components
+
+### ODPS UDF to Transform Key-value Pairs String to Multiple Values
+ODPS provides UDF for users to define transformation using Python. We choose the UDTF in UDF to implement the transformation because it can 
```

Comment:
What's UDTF short for? Maybe give links of UDF and UDTF

Reply:
Yes,  I will add the links of ODPS UDF and UDTF.